Docker image should run non-root

BlueIcarus opened the pull request:
Updated the Dockerfile to create a user and group 'Inbucket'. Set the permissions on the various files and directories. Updated the start-inbucket.sh script to not copy the greeting.html to the config directory (related to this, changed the default greeting.html file to the one in the defaults directory).
Fixes #27

jhillyerd commented:
This probably isn't documented anywhere, but the reason we copied the greeting to /config was so that it could be edited and persist between upgrades.  I'm not sure if there is a more correct way to accomplish that with Docker containers -- or if it would make more sense to hard code a default string in Inbucket for when the greeting isn't found.

BlueIcarus commented:
Actually not much has changed related to that. New users (with no configuration) will use the default `greeting.html` that is specified in the environment variable. If they wan't to use a different `greeting.html`, they can copy the one from the `defaults` directory to the `/config` directory (or any other directory) and change the environment variable to point to the new file. The only difference now is that the `greeting.html` file is not copied to the `/config` directory and that the file is therefor used from the defaults directory. For existing users, the only difference is that they will need to set the environment variable to the file in the `/config` directory (they currently have).

jhillyerd commented:
That's a good point.  Given that someone using the docker image is much less likely to poke around at the files within it, we should add some guidance to the end of the `defaults/greeting.html` on how to customize it.

You could look at the non-Docker one as starting point: https://github.com/inbucket/inbucket/blob/master/ui/greeting.html#L9  - mostly I'd want to call out the ENV variable they need to change, that they should put the new one in the /config volume so it doesn't get wiped out, and a link to a sample greeting.html on GitHub so they don't have to try and find it in the Docker image.

BlueIcarus commented:
I totally missed that this pull request was stil open. I have updated some things regarding the greetings file and updated the Dockerfile to the one that is currently in master. One of the additional changes I made is that I added the Alpine timezone data. This will allow us to set the Timezone of the image so emails will have the correct time in them when they arrive. I hope this can now be merged into master and thus a new Docker container will be release in Docker hub

jhillyerd commented on etc/docker/defaults/greeting.html:
```diff
@@ -1,17 +1,31 @@
+    soak testing your application.
+</p>
+
+<p> You can modify this greetings page by changing the Docker environment variable 'INBUCKET_WEB_GREETINGFILE'
```
Can you limit the line-length of the new paragraph to match the rest of the file above?  It looks like your editor also stripped off the trailing newline at the end of the file.

BlueIcarus replied:
As requested I split the content over more lines (including some </br> for new lines in the resulting greetings.html file). And I added the newline back to the file.

jhillyerd commented:
Looks good, thanks Martijn.  Will merge and see if there is any impact on the docker hub automated build.  I plan to cut a new beta after I finish #107
